android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -105,6 +106,131 @@ fun getProductInfo(
     }
 }
 
+fun purchase(
+    activity: Activity?,
+    options: Map<String, Any?>,
+    onResult: OnResult,
+) {
+    val purchaseParams = validatePurchaseParams(options).onFailure {
+        onResult.onError((it as PurchasesException).error.map())
+    }.getOrNull() ?: return
+    when (val purchasableItem = purchaseParams.purchasableItem) {
+        is PurchasableItem.Product -> purchaseProduct(
```

Comment:
I'm using the existing purchase functions since they all have some logic in them and it was tricky to consolidate them... The idea is to make the hybrids start using the new function, and once all hybrids have updated, we can remove them (which would be a breaking change). I just didn't want to force us to update ALL hybrids at the same time right now.